flang-rt/CMakeLists.txt

```
@@ -227,6 +227,12 @@ if (WIN32)
   find_compiler_rt_library(builtins FLANG_RT_BUILTINS_LIBRARY)
 endif ()
 
+# Build with _XOPEN_SOURCE on AIX to avoid errors caused by _ALL_SOURCE.
+# We need to enable the large-file API as well.
+if (UNIX AND ${CMAKE_SYSTEM_NAME} MATCHES "AIX")
```

```suggestion
if (UNIX AND CMAKE_SYSTEM_NAME MATCHES "AIX")
```

othwise you get strange errors when CMAKE_SYSTEM_NAME is not defined/unknown.